Gulpfile.js

```diff
@@ -105,25 +108,11 @@ function hang () {
 }
 
 // Build
-gulp.task('clean', cb => {
-    del(['./lib'], cb);
-});
-
-gulp.task('templates', ['clean'], () => {
-    return gulp
-        .src('./src/client/task.js.mustache', { silent: false })
-        .pipe(gulp.dest('./lib/client'));
+gulp.task('clean', () => {
+    return del(['./lib']);
 });
 
-gulp.task('client-scripts', ['client-scripts-bundle'], () => {
-    return gulp.src('./src/client/index.js.wrapper.mustache')
-        .pipe(mustache({ source: fs.readFileSync('./lib/client/hammerhead.js').toString() }))
-        .pipe(rename('hammerhead.js'))
-        .pipe(gulpif(!util.env.dev, uglify()))
-        .pipe(gulp.dest('./lib/client'));
-});
-
-gulp.task('client-scripts-bundle', ['clean'], () => {
+gulp.step('client-scripts-bundle', () => {
```

Comment:
I guess we can migrate to gulp 4 using a more simple manner
```js
//Gulp 3
gulp.task('client-scripts', ['client-scripts-bundle'], () => {
});

//Gulp 4
gulp.task('client-scripts', gulp.series(['client-scripts-bundle'], () => {
})); 
```

See https://github.com/gulpjs/gulp/blob/4.0/docs/API.md#gulpseriestasks

Check it and discuss with @AndreyBelym

Reply:
After discussing with @AndreyBelym we decide to use `gulp-step`.